package/MDAnalysis/core/groups.py

```diff
@@ -597,6 +597,16 @@ def __getitem__(self, item):
                 # hack to make lists into numpy arrays
                 # important for boolean slicing
                 item = np.array(item)
+
+            # correctly flatten oddly shaped, but 1d arrays
+            # raises issues if flattening already suitable arrays
+            if isinstance(item, np.ndarray) and item.ndim > 1:
+                item = np.squeeze(item)
+                # disallow high dimensional indexing.
+                # this doesnt stop the underlying issue
+                if item.ndim > 1:
+                    raise IndexError('Group index must be 1d')
+
```

Comment:
I'm not sure (?) we have to allow ndim>1, so we can probably just raise an Error inside this branch without the np.squeeze step.

Is there a code path that currently uses the `np.arange(5).reshape(-1, 1)` route?

Reply:
Not that I can tell. The only tests that fail with the sanitisation removed are the newly added tests, to test the sanitisation is working.

Reply:
Ok great then, I couldn't remember if something like Residue access was going via this route.